refactor/refactor_test.go

```diff
@@ -337,7 +337,7 @@ p = 7`)
 		t.Fatal("Expected error but got nil")
 	}
 
-	errMsg := "rego_type_error: conflicting rules named p found"
+	errMsg := "rego_type_error: conflicting rules data.b.p found"
```

Comment:
Talking about "named" rules has become more difficult: is the name of
```
p.q.r[x] = 10 if x := input.x
```
`p` or `p`? So we're giving the full ref here instead.